### PR TITLE
Silently ignore the obsolete `sudo` token scope

### PR DESCRIPTION
**Pull request by earl-warren**
Fixes: https://codeberg.org/forgejo/forgejo/issues/820

### CODE REVIEW

File: models/auth/token_scope.go [hunk 1]
```diff
@@ -250,7 +250,7 @@ func (s AccessTokenScope) parse() (accessTokenScopeBitmap, error) {
 			remainingScopes = remainingScopes[i+1:]
 		}
 		singleScope := AccessTokenScope(v)
-		if singleScope == "" {
+		if singleScope == "" || singleScope == "sudo" {
```
Comment (delvh) [inline models/auth/token_scope.go]:
```suggestion
		if singleScope == "" || singleScope == "sudo" { // sudo was recognized previously but isn't anymore
```

**Review by delvh — DISMISSED** [review]

**Review by techknowlogick — CHANGES_REQUESTED** [review]
I'm unsure of this change, as it silently hides an error. Users should be alerted if they are using incorrect scopes even if it were previously accepted. By hiding the error it adds the possibility that users are expecting sudo to work, or introduce hidden issues with external tools.

**Comment by jackHay22** [comment]
Why is `sudo` a special case here? The `sudo` scope is included in the migration that updates the scopes for existing tokens. Is there a reason that only `sudo` needs to be ignored?

**Comment by delvh** [comment]
If I had to guess, https://discord.com/channels/322538954119184384/322910365237248000/1116339032033603584 encountered the same problem.
So, my guess is there's a bug in the migration regarding `sudo` scopes?

**Comment by jackHay22** [comment]
> So, my guess is there's a bug in the migration regarding sudo scopes?

My understanding of the issue is that a user tried to create a new access token using the `sudo` scope which is no longer supported. The same logic could be applied to any of the old scopes that are no longer supported. It's not clear to me that this is an issue with the migration.

File: models/auth/token_scope.go (hunk 1 above)
Comment (jackHay22) [reply models/auth/token_scope.go]: By this logic, wouldn't we be ignoring all previous scopes that we no longer support?

**Pull request closed without merging**